### ISSUE TITLE
🐛 BUG: Prettier is expanding self closing empty tags into its full form

### ISSUE THREAD

**Issue by ni554n** [withastro/prettier-plugin-astro#325]
### Describe the Bug

Prettier is unnecessarily turning any `<svg />` like tags into `<svg></svg>` in a .astro file.
Only default HTML elements are affected; custom ones are fine.

### Steps to Reproduce

https://stackblitz.com/edit/node-dyqbs8?file=src%2Fpages%2Findex.astro,src%2Fcomponents%2FCustom.astro,package.json

> Open _src/pages/index.astro_ and run `npm run format`

**Comment by Princesseuh** [comment]
Hello, this is intended, we follow HTML rules in this regard, so `svg` cannot be a self closing tag.

Is this causing any issues in your workflow?

**Comment by ni554n** [comment]
Yes, I'm using Solid's ESLint plugin which has this [self-closing-comp](https://github.com/solidjs-community/eslint-plugin-solid/blob/main/docs/self-closing-comp.md) rule enabled. I can turn it off but I prefer it in that way.

Is there any specific reason why we are enforcing this HTML rule in JSX? Prettier itself doesn't enforce this by default.

**Comment by Princesseuh** [comment]
> Yes, I'm using Solid's ESLint plugin which has this [self-closing-comp](https://github.com/solidjs-community/eslint-plugin-solid/blob/main/docs/self-closing-comp.md) rule enabled. I can turn it off but I prefer it in that way.
> 
> Is there any specific reason why we are enforcing this HTML rule in JSX? Prettier itself doesn't enforce this by default.

Astro in general does not follow JSX rules, even outside of the additional features we have, we also have HTML whitespace sensitivity, unlike JSX for instance. However, it seems like Prettier also doesn't enforce this inside HTML files, so we shouldn't be either

It's a bit unfortunate though, as it mean the plugin would allow invalid HTML 😦

**Comment by ni554n** [comment]
We should have a flag to disable this rule then.
SVG elements such as _path_, _circle_, _filters_ are naturally empty and self-closing. Here's a MDN example of [path element](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/path).

**Comment by aadito123** [comment]
> > Yes, I'm using Solid's ESLint plugin which has this [self-closing-comp](https://github.com/solidjs-community/eslint-plugin-solid/blob/main/docs/self-closing-comp.md) rule enabled. I can turn it off but I prefer it in that way.
> > Is there any specific reason why we are enforcing this HTML rule in JSX? Prettier itself doesn't enforce this by default.
> 
> Astro in general does not follow JSX rules, even outside of the additional features we have, we also have HTML whitespace sensitivity, unlike JSX for instance. However, it seems like Prettier also doesn't enforce this inside HTML files, so we shouldn't be either
> 
> It's a bit unfortunate though, as it mean the plugin would allow invalid HTML frowning

I think at least the self-closing SVG elements should be allowed since that would be entirely valid HTML.

**Comment by Princesseuh** [comment]
Hi everyone! Submitted a PR fixing this: https://github.com/withastro/prettier-plugin-astro/pull/328, however it is currently blocked by a compiler issue, so we'll have to wait on that. Sorry for the inconvenience!
